main.go

```diff
@@ -242,6 +247,27 @@ func main() {
 	logger.Info().Msgf("Terminating from signal: %s", sig)
 }
 
+func getVehicleInfo(err error, logger zerolog.Logger, ethAddr *common.Address) *models.VehicleInfo {
+	identityAPIService := gateways.NewIdentityAPIService()
+	var vehicleDefinitions *[]models.VehicleInfo
+	vehicleDef, err := gateways.Retry(3, 1*time.Second, logger, func() (interface{}, error) {
+		return identityAPIService.QueryIdentityAPIForVehicles(ethAddr.Hex())
+	})
+
+	if vehicleDef != nil {
+		definitions := vehicleDef.([]models.VehicleInfo)
+		vehicleDefinitions = &definitions
+	} else {
+		logger.Err(err).Msg("failed to get vehicle definitions")
+	}
+
+	var firstVehicleDefinition *models.VehicleInfo
+	if vehicleDefinitions != nil && len(*vehicleDefinitions) > 0 {
+		firstVehicleDefinition = &(*vehicleDefinitions)[0]
```

Comment:
I naively get the first element, but I'm afraid that's probably not right. Please suggest.